Update dependency `handlebars` to v6 for `mdman`

### Problem

Renovate bot found a newer version available for dependency `handlebars` in <https://github.com/rust-lang/cargo/pull/14763>. `handlebars` is used by `mdman` for templatizing and generating man pages.

However, the new version has some API changes that need to be fixed:


<details><summary>Build failure. Click to See details</summary>
<p>

```
   Compiling mdman v0.0.0 (/Users/runner/work/cargo/cargo/crates/mdman)
error[E0639]: cannot create non-exhaustive struct using struct expression
  --> crates/mdman/src/hbs.rs:27:19
   |
27 |       let options = DirectorySourceOptions {
   |  ___________________^
28 | |         tpl_extension: ".md".to_string(),
29 | |         hidden: false,
30 | |         temporary: false,
31 | |     };
   | |_____^
```

</p>
</details> 

### Steps

1. Clone `rust-lang/cargo`
2. Run `cargo build-man`
3. Fix the build failure.

### Possible Solution(s)

Look into what have been changed in handlebars and update accordingly.

### Version

```text
7421ccf9159d5ac43883c53abe41ef1da657acfd
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -50,7 +50,7 @@ git2 = "0.19.0"
 git2-curl = "0.20.0"
 gix = { version = "0.67.0", default-features = false, features = ["blocking-http-transport-curl", "progress-tree", "parallel", "dirwalk"] }
 glob = "0.3.1"
-handlebars = { version = "5.1.2", features = ["dir_source"] }
+handlebars = { version = "6.0.0", features = ["dir_source"] }
 hex = "0.4.3"
 hmac = "0.12.1"
 home = "0.5.9"

--- a/crates/mdman/src/hbs.rs
+++ b/crates/mdman/src/hbs.rs
@@ -24,11 +24,8 @@ pub fn expand(file: &Path, formatter: FormatterRef<'_>) -> Result<String, Error>
     handlebars.register_decorator("set", Box::new(set_decorator));
     handlebars.register_template_file("template", file)?;
     let includes = file.parent().unwrap().join("includes");
-    let options = DirectorySourceOptions {
-        tpl_extension: ".md".to_string(),
-        hidden: false,
-        temporary: false,
-    };
+    let mut options = DirectorySourceOptions::default();
+    options.tpl_extension = ".md".to_string();
     handlebars.register_templates_directory(includes, options)?;
     let man_name = file
         .file_stem()

